src/hera.cpp

```diff
@@ -405,8 +405,15 @@ int hera_set_option(
   }
 
   if (strcmp(name, "engine") == 0) {
-    if (wasm_engine_options.count(value)) {
-      hera->wasm_engine = wasm_engine_options.at(value);
+    auto it = wasm_engine_options.find(value);
+    if (it != wasm_engine_options.end()) {
+      switch (it->second) {
+      case hera_wasm_engine::binaryen:
+        hera->engine.reset(new BinaryenEngine);
```

Comment:
If we're keeping the enum then I'd create a helper closer where the enum is defined: `hera_create_engine(hera_instance* hera, string const& name);`

Reply:
@chfast I made this change. Merging it this way, because I am not sure wabt/wavm will not need instance variables. Once we have a basic wabt/wavm code merged which works with this limitation I'd be happy to remove this extra indirection of having `enum hera_wasm_engine`.